protocol/da-footprint-block-limit.md

```diff
@@ -0,0 +1,299 @@
+# Jovian: DA Footprint Block Limit
+
+|                    |                                                    |
+| ------------------ | -------------------------------------------------- |
+| Author             | _Sebastian Stammler_                                 |
+| Created at         | _2025-08-12_                                         |
+| Initial Reviewers  | _TBD_                 |
+| Need Approval From | _TBD_                                    |
+| Status             | _Draft<!--/ In Review / Implementing Actions / Final-->_ |
+
+## Purpose
+
+Evaluate proposal to add a DA footprint block limit to mitigate DA spam and prevent priority fee auctions from occurring.
+This proposal is an alternative to [Custom Calldata Floor Gas](https://github.com/ethereum-optimism/design-docs/pull/294),
+avoiding the need to change individual transaction gas mechanics at all.
+
+## Summary
+
+A DA footprint block limit is introduced to mitigate DA spam and prevent priority fee auctions. By tracking DA footprint
+of transactions alongside gas, this approach adjusts the block gas limit to account for high estimated DA usage without
+altering individual transaction gas mechanics. Preliminary analysis shows minimal impact on most blocks on production
+networks like Base or OP Mainnet.
+
+## Problem Statement + Context
+
+<!-- Describe the specific problem that the document is seeking to address as well
+as information needed to understand the problem and design space.
+If more information is needed on the costs of the problem,
+this is a good place to that information. -->
+
+The current L1 cost model doesn't account for *limited L1 data availability (DA) throughput*. 
+This can lead to network congestion for batchers if the throughput on L2s surpasses that of L1 blob space, or if blob
+space is congested for other reasons.
+
+Current L1 throughput is 6 blobs per block at target with a max of 9 blobs per block.
+This is approx. `128 kB * 6 / 12 s = 64 kB/s` or `96 kB/s`, resp. The L1 cost is proportional to the L1 origin’s base
+fee and blob base fee. But Most OP Stack chains have gas targets high enough to accept far more than 128kb of calldata
+per (2s) block. So the calldata floor cost doesn't nearly limit calldata throughput enough to prevent.
```

Comment:
Trails off here. I think you mean priority fee auctions. Recommend introducing PFA acronym.